src/ninja.cc

```diff
@@ -37,15 +37,12 @@
 #include "public/tools.h"
```

Comment:
Not a comment on the way you implemented this change, just a comment on the concept.

I could see a benefit to having the build system set up to allow each of these tools to be their own binary. That would allow people who are compiling the Ninja project to choose a multi-binary, or a set of stand alone binaries (and both options would also allow dynamic or static linking against the library-fied libninja)